src/qililab/platform/platform.py

```diff
@@ -627,18 +635,29 @@ def execute_anneal_program(
             transpiler (Callable): ising to flux transpiler. The transpiler should take 2 values as arguments (delta, epsilon) and return 2 values (phix, phiz)
             averages (int, optional): Amount of times to run and average the program over. Defaults to 1.
         """
-        annealing_program = AnnealingProgram(self, annealing_program_dict)
-        annealing_program.transpile(transpiler)
-        annealing_waveforms = annealing_program.get_waveforms()
-
-        qp_annealing = QProgram()
-        with qp_annealing.average(averages):
-            for bus, waveform in annealing_waveforms.values():
-                qp_annealing.play(bus=bus.alias, waveform=waveform)
-
-        # TODO: define readout
-
-        self.execute_qprogram(qprogram=qp_annealing)
+        if calibration and calibration.has_waveform(bus=readout_bus, name=measurement_name):
```

Comment:
why have `if calibration` instead of making calibration non-optional?

Reply:
Rigth again, at the beginning we started with calibration non optional, but it is a must now for annealing, changing it.